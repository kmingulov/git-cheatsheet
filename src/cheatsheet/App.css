.header {
  padding-left: 15px !important;
  padding-right: 15px !important;
}

.mainContent {
  padding-top: 20px;
}

.git-command-column {
  width: 40%;
}

/* .git { } */

/* .git-command { } */

/* .git-remote { } */

.git-url {
  text-decoration: underline;
}

.git-branch {
  font-weight: bold;
}

/* .git-commit { } */

/* .git-ref { } */

.git-file {
  font-style: italic;
}
