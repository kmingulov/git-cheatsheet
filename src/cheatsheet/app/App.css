.header {
  padding-left: 15px !important;
  padding-right: 15px !important;
}

.mainContent {
  padding-top: 20px;
}

.searchBox {
  padding-bottom: 20px;
}
