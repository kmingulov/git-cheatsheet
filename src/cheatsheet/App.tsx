import React, { Component, ReactElement } from 'react';
import { Container, FormControl, InputGroup, Navbar } from 'react-bootstrap';
import { FaSearch } from 'react-icons/fa';

import { CommandStore } from './command-store';
import { CommandTable } from './command-table';

import commands from './command-data/commands';

import './App.css';

const store = new CommandStore(commands);

const initialState = { searchTerm: '' };
type State = Readonly<typeof initialState>;

class App extends Component<object, State> {
  public readonly state: State = initialState;

  public render(): ReactElement {
    return (
      <div>
        <Navbar bg='primary' sticky='top'>
          <Container className='header'>
            <Navbar.Brand>Git Cheatsheet</Navbar.Brand>
          </Container>
        </Navbar>

        <Container className='mainContent'>
          <InputGroup className='searchBox'>
            <InputGroup.Prepend>
              <InputGroup.Text>
                <FaSearch/>
              </InputGroup.Text>
            </InputGroup.Prepend>
              <FormControl
                placeholder='Searchâ€¦'
                onChange={ (event: any) => this.setState({ searchTerm: event.target.value }) }
              />
          </InputGroup>

          <CommandTable store={ store } searchTerm={ this.state.searchTerm }/>
        </Container>
      </div>
    );
  }
}

export default App;
