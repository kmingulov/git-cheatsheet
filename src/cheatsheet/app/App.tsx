import React, { Component, ReactElement } from 'react';
import { Container } from 'react-bootstrap';

import { AppHeader } from './AppHeader';

import { AboutModal } from './modals';
import { CommandPage } from './pages';

import './App.css';

type State = Readonly<{ aboutModalShown: boolean }>;

/**
 * Main application component.
 * @inheritdoc
 */
export class App extends Component<{}, State> {
  public state: State = { aboutModalShown: false };

  public render(): ReactElement {
    return (
      <>
        <AboutModal shown={ this.state.aboutModalShown } onClose={ this.hideAbout }/>
        <AppHeader onAbout={ this.showAbout }/>
        <Container className='mainContent'>
          <CommandPage/>
        </Container>
      </>
    );
  }

  private showAbout = () => {
    this.setState({ aboutModalShown: true });
  }

  private hideAbout = () => {
    this.setState({ aboutModalShown: false });
  }
}
