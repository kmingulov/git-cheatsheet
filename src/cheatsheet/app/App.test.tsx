import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

import { App } from './App';

jest.mock('./AppHeader');
jest.mock('./modals/AboutModal');
jest.mock('./pages/CommandPage');

Enzyme.configure({adapter: new Adapter()});

describe(App, () => {
  it('renders correctly', () => {
    const component = mount(<App/>);
    expect(component).toMatchSnapshot();
  });
});
