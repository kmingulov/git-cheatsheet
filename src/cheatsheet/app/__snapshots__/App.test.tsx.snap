// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<App>
  <div>
    <AppHeader />
    <ForwardRef(Bootstrap(Container))
      className="mainContent"
    >
      <Container
        as="div"
        bsPrefix="container"
        className="mainContent"
        fluid={false}
      >
        <div
          className="mainContent container"
        >
          <CommandPage />
        </div>
      </Container>
    </ForwardRef(Bootstrap(Container))>
  </div>
</App>
`;
